### DISCUSSION TITLE
DefaultStylePicker and block.json isDefault style strange behaviour. Bug or feature?

### DISCUSSION THREAD

**Discussion by realthemes** [WordPress/gutenberg#52855] [Q&A] [answered]
Is this how it should work?
I made a custom block with create-block, just a simple examle block that return paragraph, then in block.json I add fields:
```
"supports": {
    "defaultStylePicker": true
},
"styles": [{
	"isDefault": true,
	"label": "Primary style",
	"name": "primary"
}, {
	"label": "Secondary style",
	"name": "secondary"
}],
```
But when I add my block on page, the default classname `is-style-primary` didn't applied until I press Default button in Editor Sidebar, in its turn, Default button looks like it already pressed and if you click it, you can't unpress it.
It turns out that Default styles need to be duplicated – one set of styles for the initial block added to the page, the same styles for 'isDefault,' and a third set of styles for 'Secondary.'
What you think it's bug or feature?

**Comment by sirreal** [comment, marked as answer]
Yes, this does appear to be a bug. There's a report of this behavior in https://github.com/WordPress/gutenberg/issues/38890.

Closing as duplicate.
